edi_storage/components/listener.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from pathlib import PurePath
+
+from odoo.addons.component.core import Component
+
+
+class EdiStorageListener(Component):
+    _name = "edi.storage.component.listener"
+    _inherit = "base.event.listener"
+
+    def _move_file(self, storage, from_dir_str, to_dir_str, filename):
+        from_dir = PurePath(from_dir_str)
+        to_dir = PurePath(to_dir_str)
+        try:
+            storage.move_files([(from_dir / filename).as_posix()], to_dir.as_posix())
```

Comment:
it would better to do this after the transaction is done. 
Please find an example here https://github.com/OCA/connector-interfaces/pull/74/files#diff-60319609653ac5e93d1758d594b36b9274e778933247e17ef51799b7a28a341bR61 (BTW if you have time to drop a review there it would be nice :wink: )

Reply:
Interesting :thinking: I will need to check if this is compatible with trying twice to move files, first from pending and then from error. 

BTW, tests added.

Reply:
Done, the changes is here: https://github.com/OCA/edi/pull/408/commits/efc487e8d5bd39f87589bd0e89ad2705c8bc0d59.

Reply:
could you drop a review there when you time? :pray:

Reply:
It is a big PR but I will try to review it this week! :smile:

Reply:
actually part of it is from another PR that should be reviewed first. BTW that PR adds kind of the same concept we have here in EDI module to allow advanced configuration for components.